Yosemite/Yosemite/Actions/ProductTagAction.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+import Networking
+
+/// ProductTagAction: Defines all of the Actions supported by the ProductTagStore.
+///
+public enum ProductTagAction: Action {
+
+    /// Synchronizes all ProductTags matching the specified criteria.
+    /// `onCompletion` will be invoked when the sync operation finishes.
+    ///
+    case synchronizeAllProductTags(siteID: Int64, fromPageNumber: Int, onCompletion: (ProductTagActionError?) -> Void)
```

Comment:
I'd say that this method is not born for being used with the `SyncingCoordinator`, since the implementation works to avoid it. We want to fetch all the tags, hiding the pagination. `fromPageNumber` is here because I use it to recover the pagination from a page in the case at a certain point the fetching fails. Using this page here, it's useful to avoid errors.

> also, would be nice if we can use Result<[ProductTag], Error> in the completion block!

This is a particular case, where explicitly I didn't use `Result<>`, because here I just need a `ProductTagActionError` as completion in case the synchronization failed.  Using `ProductTagActionError`, I can recover the synchronization from a specified `pageNumber` (as mentioned above).

Reply:
this makes sense now since we aren't using pagination for this action 👍